src/python/pants/backend/python/dependency_inference/module_mapper.py

```diff
@@ -62,17 +65,24 @@ class FirstPartyPythonMappingImplMarker:
     """
 
 
-class FirstPartyPythonModuleMapping(FrozenDict[str, Tuple[Address, ...]]):
+@dataclass(frozen=True)
+class FirstPartyPythonModuleMapping:
     """A merged mapping of module names to owning addresses.
 
     This mapping may have been constructed from multiple distinct implementations, e.g.
     implementations for each codegen backends.
     """
 
-    def addresses_for_module(self, module: str) -> Tuple[Address, ...]:
-        addresses = self.get(module)
-        if addresses:
-            return addresses
+    mapping: FrozenDict[str, tuple[Address, ...]]
+    ambiguous_modules: FrozenDict[str, tuple[Address, ...]]
+
+    def addresses_for_module(self, module: str) -> tuple[tuple[Address, ...], tuple[Address, ...]]:
+        """Return all unambiguous and ambiguous addresses."""
```

Comment:
A comment explaining here that we might have two unambiguous `Address`es for one module due to type stubs would be helpful.